roles/addons/diskless/files/livenet_module.py

```diff
@@ -139,58 +139,56 @@ def generate_file_system(self):
     def generate_operating_system(self):
         logging.info('Generating image \'' + self.name + '\' operating system')
 
-        # create image personnal working directory
+        # Create os generation directory
         logging.debug('Executing \'mkdir -p ' + self.WORKING_DIRECTORY + 'generated_os\'')
         os.makedirs(self.WORKING_DIRECTORY + 'generated_os')
 
+        # Set up release version
         if hasattr(self, 'release_version'):
-            release = '--releasever=' + self.release_version
+            release = ' --releasever=' + self.release_version
         else:
             release = ''
 
-        # Populating dnf packages list
-        dnf_packages_list = []
+        # Create empty string of packages
+        dnf_packages = ''
 
-        # Generate desired image file system
+        # Get appropriate packages for desired image file system
         if self.livenet_type == LivenetImage.Type.STANDARD:
             logging.debug('Standard image requested. Adding "@core" to packages list')
-            dnf_packages_list.append('@core')
+            dnf_packages = '@core '
+
         elif self.livenet_type == LivenetImage.Type.SMALL:
             logging.debug('Small image requested. Adding "dnf yum iproute procps-ng openssh-server NetworkManager" to packages list')
-            dnf_packages_list.append('dnf yum iproute procps-ng openssh-server NetworkManager')
+            dnf_packages = 'dnf yum iproute procps-ng openssh-server NetworkManager '
+
         elif self.livenet_type == LivenetImage.Type.CORE:
             logging.debug('Core image requested. Adding "iproute procps-ng openssh-server" to packages list')
-            dnf_packages_list.append('iproute procps-ng openssh-server')
+            dnf_packages = 'iproute procps-ng openssh-server '
 
         # If there are additional packages to install
         if hasattr(self, 'additional_packages'):
+            logging.debug('Additional packages requested. Adding "' + str(self.additional_packages) + '" to dnf packages to install')
 
             # Add additional packages to the dnf packages list
-            packages = ''
             for package in self.additional_packages:
-                packages = packages + ' ' + package
-            logging.debug('Additional packages requested. Adding "' + packages + '" to packages list')
-            dnf_packages_list.append(packages)
+                dnf_packages = dnf_packages + package + ' '
```

Comment:
Since `self.additional_packages` is a list, you can avoid the for-loop and use join() instead:
`dnf_packages += ' ' + ' '.join(self.additional_packages)`.